src/webrtc.ts

```diff
@@ -0,0 +1,482 @@
+      sdpSemantics: 'unified-plan',
+      iceTransportPolicy: this.config.iceTransportPolicy || 'all'
+    });
+  }
+
+  prepareSession (options: any) {
+    options.iceServers = this.config.iceServers || options.iceServers;
+    options.optOutOfWebrtcStatsTelemetry = !!this.config.optOutOfWebrtcStatsTelemetry;
+
+    const session = new GenesysCloudMediaSession(options, this.getSessionTypeByJid(options.peerID), this.config.allowIPv6);
+    this.proxyStatsForSession(session);
+    return session;
+  }
+
+  // This should probably go into the webrtc sdk, but for now I'm putting here so it's in a central location.
+  // This should be moved when the sdk is the primary consumer
+  proxyStatsForSession (session: GenesysCloudMediaSession) {
+    session.on('stats', (statsEvent: StatsEvent) => {
+      const extraDetails = {
+        conference: (session as any).conversationId,
+        session: session.sid,
+        sessionType: session.sessionType
+      };
+
+      // format the event to what the api expects
+      const event = formatStatsEvent(statsEvent, extraDetails);
+
+      this.statsToSend.push(event);
+      this.throttledSendStats();
+    });
+  }
+
+  async sendStats () {
+    const stats = this.statsToSend.splice(0, this.statsToSend.length);
+
+    if (!stats.length) {
+      return;
+    }
+
+    const data = {
+      appName: 'streamingclient',
+      appVersion: Client.version,
+      actions: stats
+    };
+
+    // At least for now, we'll just fire and forget. Since this is non-critical, we'll not retry failures
+    try {
+      await requestApi('diagnostics/newrelic/insights', {
+        method: 'post',
+        host: this.client.config.apiHost,
+        authToken: this.client.config.authToken,
+        logger: this.client.logger,
+        data
+      });
+    } catch (err) {
+      this.logger.error('Failed to send stats', { err, stats });
+    }
+  }
+
+  addEventListeners () {
+    this.client.on('connected', async () => {
+      await this.refreshIceServers();
+    });
+
+    this.client._stanzaio.jingle.on('log', (level, message, data?) => {
+      this.logger[level](message, data);
+    });
+
+    this.client._stanzaio.on('message', (msg: any) => {
+      if (msg.propose) {
+        this.handlePropose(msg);
+      }
+    });
+  }
+
+  proxyEvents () {
+    // this.jingleJs.on('send', data => {
+    //   if (data.jingle && data.jingle.sid && this.ignoredSessions.get(data.jingle.sid)) {
+    //     this.logger.debug('Ignoring outbound stanza for ignored session', data.jingle.sid);
+    //     return;
+    //   }
+    //   this.emit('send', data);
+    // });
+
+    this.client._stanzaio.on('jingle:outgoing', session => {
+      return this.emit(events.OUTGOING_RTCSESSION, session);
+    });
+
+    this.client._stanzaio.on('jingle:incoming', session => {
+      return this.emit(events.INCOMING_RTCSESSION, session);
+    });
+
+    const eventsToProxy = [
```

Comment:
Since we are using a strict event emitter, it may be nice to have these events strongly typed. I think something like this will work: 
``` ts
const eventsToProxy: Array<keyof SessionEvents> = [ ... ];
```